### ISSUE TITLE
PropType validation for stateless function components isn't working from 3.6.0 release

### ISSUE THREAD

**Issue by trevordmiller** [yannickcr/eslint-plugin-react#268]
PropType validation for stateless function components isn't working from 3.6.0 release - example below. I'm really looking forward to this feature, thank you for all your hard work.

![proptype-validation-stateless-components](https://cloud.githubusercontent.com/assets/5497885/10644949/66223d3e-77e8-11e5-8540-c57365d240e3.png)

**Comment by luiscarli** [comment]
I'm also having a prop-type check problem in my stateless functions.
In my case when I have a function with a implicit return there's no warning for missing propTypes.
For example:

```jsx
// no react/prop-types warning in this case
const Annotation = props => (
  <div>
    {props.text}
  </div>
)
```
